### ISSUE TITLE
Request to have upload path be configurable. 

### ISSUE THREAD

**Issue by briancooper** [ryanto/next-s3-upload#6]
More of a feature request. 

Would like to have this be able to be configurable 
let key = `next-s3-uploads/${uuidv4()}/${filename.replace(/\s/g, '-')}`;

Where next-s3-uploads could be passed-in or from process.env.S3_BASE_PATH
Something like that. 

Thanks and it is working great.

**Comment by ryanto** [comment]
Great suggestion! 

I'm thinking this should be added via the API file rather than having to set another env var. Something a long the lines of...

```js
// pages/api/s3-upload.js
import APIRoute from 'next-s3-upload';

export default APIRoute({ basePath: "/my/base/path" })
```

Let me know if that would work for you.

**Comment by coopemor** [comment]
Would it be too hard to make it as an option on the function call? I could see something like this being useful:

```
uploadToS3(file, {
      basePath: "/my/base/path"
})
```

**Comment by coopemor** [comment]
I retract it, from a security perspective makes more sense to have multiple endpoints with different paths! Your solution makes sense to me.

**Comment by ryanto** [comment]
Ok this is all set, it's available in version `0.1.4`.

The way it works is by providing a custom key function to the API route. Here's an example:

```js
// pages/api/s3-upload.js
import { APIRoute } from 'next-s3-upload';

export default APIRoute.configure({
  key(req, filename) {
    return `my/uploads/path/${filename.toUpperCase()}`;
  },
});
```

The key function gets both the API request as well as the file name as arguments. It can return any string and that string will be used as the full path s3 key.

Let me know if you run into any trouble with this!

**Comment by priyaalba** [comment]
> Ok this is all set, it's available in version `0.1.4`.
> 
> The way it works is by providing a custom key function to the API route. Here's an example:
> 
> ```js
> // pages/api/s3-upload.js
> import { APIRoute } from 'next-s3-upload';
> 
> export default APIRoute.configure({
>   key(req, filename) {
>     return `my/uploads/path/${filename.toUpperCase()}`;
>   },
> });
> ```
> 
> The key function gets both the API request as well as the file name as arguments. It can return any string and that string will be used as the full path s3 key.
> 
> Let me know if you run into any trouble with this!

If we change the path, it throws access denied in .js

**Comment by ronateds** [comment]
> > Ok this is all set, it's available in version `0.1.4`.
> > The way it works is by providing a custom key function to the API route. Here's an example:
> > ```js
> > // pages/api/s3-upload.js
> > import { APIRoute } from 'next-s3-upload';
> > 
> > export default APIRoute.configure({
> >   key(req, filename) {
> >     return `my/uploads/path/${filename.toUpperCase()}`;
> >   },
> > });
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The key function gets both the API request as well as the file name as arguments. It can return any string and that string will be used as the full path s3 key.
> > Let me know if you run into any trouble with this!
> 
> If we change the path, it throws access denied in .js

I was having access denied using this function.

```
// pages/api/s3-upload.js
import { APIRoute } from "next-s3-upload";

export default APIRoute.configure({
  key(req, filename) {
    return `my/uploads/path/${filename.toUpperCase()}`;
  }
});
```

but the async function worked for me.

```
// pages/api/s3-upload.js
import { APIRoute } from "next-s3-upload";

export default APIRoute.configure({
  async key(req, filename) {
    let path = "somepath"
    return `${path}/${filename}`;
  }
});
```

**Comment by geraldinefiser** [comment]
Hi, 

I am getting the same access denied issue whether or not I use the async function (see below). 
I confirmed that I get no issue uploading when using the default API route without customizing the path. 
Any ideas on how I can fix that?

`
import { APIRoute } from "next-s3-upload";
export default APIRoute.configure({
  async key(req, filename) {
    let contractType = req.body.contractType;
    
    return `${contractType}/${uuidv4()}`;
  }
});`
